File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    flag = [0 for _ in range(10)]
    # [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
    is_repeat = False
    
    answer.append(arr[0])
    for i in range(1,len(arr)):
        if arr[i]==arr[i-1]:
            is_repeat=True
        if arr[i] not in answer:
            answer.append(arr[i]) #answer에 없으면 무조건 삽입
        elif arr[i] in answer and is_repeat==False:
            answer.append(arr[i])
        is_repeat=False
            
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        // int[] answer = {};
        ArrayList<Integer> answer = new ArrayList<>();
        
        answer.add(arr[0]);
        for (int i=1; i<arr.length; i++) {
            if (arr[i-1]!=arr[i])
                answer.add(arr[i]);
            else continue;
        }

        return answer.stream().mapToInt(i->i).toArray();
    }
}